### DISCUSSION TITLE
Global Data Type Components

### DISCUSSION THREAD

**Discussion by ghislaineguerin** [mathesar-foundation/mathesar#613] [Maintainer discussions]
Design for supporting data types functionality was delivered in parts, with design increments targeting one data type at the time, however, all parts must be implemented cohesively. To avoid issues with implementation, the following document goes over the design for data type settings, filtering, grouping, sorting, and input and corrects some issues uncovered during the design for the ['Initial Data Types'](https://github.com/centerofci/mathesar/milestone/11) milestone. 

## Document
https://wiki.mathesar.org/en/design/specs/global-data-type-components

@pavish Your feedback is not required, but you might have some comments from a front-end perspective. Please take a look and let me know.
@kgodey @mathemancer no feedback required. Also, feel free to add or suggest improvements.

**Comment by kgodey** [comment]
@ghislaineguerin Looks good, this must have been a lot of work to put together but it's nice to see everything in one place. The only thing I see missing was using tabular figures/monospace to display numbers.
